Panic in Node Exporter on v0.37.0-rc0

### What's wrong?

I am saying a Panic in the Agent, running the v0.37.0-rc.0 for what appears to be the Node Exporter.

I have updated my Node Exporter Configuration with the changes for this version (adding a Label), but everytime its scrapped (I think) its causing a Panic

### Steps to reproduce

Deploy Grafana Agent v0.37.0-rc.0
Use the Configuration below
Run the Agent
Leave for ~2 mins
Observe Panic

### System information

Linux, Agent running in Docker

### Software version

v0.37.0-rc.0

### Configuration

```text
// Config: Logging

logging {
  // https://grafana.com/docs/agent/latest/flow/reference/config-blocks/logging/
  level  = coalesce(env("GRAFANA_AGENT_LOG_LEVEL"), "info")
  format = "logfmt"
}

// Config: Tracing

tracing {
  // https://grafana.com/docs/agent/latest/flow/reference/config-blocks/tracing/
  sampling_fraction = 1
  write_to = [otelcol.processor.attributes.default.input]
}

// Discover Docker Containers

discovery.docker "docker_containers" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/discovery.docker/
  host = "unix:///var/run/docker.sock"
  refresh_interval = "30s"
}

discovery.relabel "docker_containers" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/discovery.relabel/
  targets = discovery.docker.docker_containers.targets
  rule {
    action        = "replace"
    source_labels = ["__address__"]
    target_label  = "container_address"
    regex         = "([^:]+)(:[0-9]+)?"
    replacement   = "${1}"
  }
  rule {
    action        = "replace"
    source_labels = ["__meta_docker_container_id"]
    target_label  = "container_id"
  }
  rule {
    action        = "replace"
    source_labels = ["__meta_docker_container_name"]
    target_label  = "pod"
    regex         = "/(.*)"
  }
  rule {
    action        = "replace"
    source_labels = ["pod"]
    target_label  = "app"
    regex         = "(.+?)(-\\d+)?"
    replacement   = "${1}"
  }
  rule {
    source_labels = ["app"]
    target_label  = "container"
  }
  rule {
    action        = "replace"
    source_labels = ["pod","__meta_docker_port_private"]
    target_label  = "instance"
    separator     = ":"
  }
  rule {
    action        = "replace"
    source_labels = ["__meta_docker_container_label_namespace","__meta_docker_container_label_com_docker_compose_service"]
    target_label  = "job"
    separator     = "/"
  }
  rule {
    action        = "replace"
    source_labels = ["__meta_docker_container_label_namespace","container"]
    target_label  = "job"
    separator     = "/"
  }
  rule {
    action        = "replace"
    source_labels = ["__meta_docker_container_label_namespace"]
    target_label  = "namespace"
  }
  rule {
    action        = "replace"
    source_labels = ["__meta_docker_container_label_com_docker_compose_service"]
    target_label  = "service"
  }
}

// Faro Receiver

faro.receiver "default" {
  // https://grafana.com/docs/agent/next/flow/reference/components/faro.receiver/
  server {
    listen_address       = "0.0.0.0"
    listen_port          = 12347
    cors_allowed_origins = ["*"]
  }

  extra_log_labels = {
    "source" = "faro",
  }

  sourcemaps {
    download = true
  }

  output {
    logs   = [loki.process.faro.receiver]
    traces = [otelcol.processor.attributes.default.input]
  }
}

// Metrics
// Docs (Scrape): https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.scrape/
// Docs (Relabel): https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.relabel/
// Docs (Remote Write): https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.remote_write/

prometheus.exporter.agent "host" {
  // https://grafana.com/docs/agent/next/flow/reference/components/prometheus.exporter.agent/
}

prometheus.exporter.cadvisor "docker" {
  // https://grafana.com/docs/agent/next/flow/reference/components/prometheus.exporter.cadvisor/
  docker_host      = "unix:///var/run/docker.sock"
  storage_duration = "1m"
}

prometheus.exporter.mysql "grafanadb" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.mysql/
  data_source_name  = "root:" + env("GRAFANA_DB_MARIADB_ROOT_PASSWORD") + "@(" + env("GRAFANA_DB_CONTAINER_NAME") + ":3306)/"
  enable_collectors = ["engine_innodb_status", "heartbeat", "info_schema.clientstats", "mysql.user"]
}

prometheus.exporter.redis "grafana_caching" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.redis/
  redis_addr = env("GRAFANA_REDIS_CONTAINER_NAME") + ":6379"
}

prometheus.exporter.unix "host" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.unix/
  include_exporter_metrics = true
  rootfs_path = "/rootfs"
}

prometheus.scrape "agent_exporter" {
  // Configure a prometheus.scrape component to collect agent_exporter metrics.
  targets    = prometheus.exporter.agent.host.targets
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
}

prometheus.scrape "cadvisor_exporter" {
  // Configure a prometheus.scrape component to collect cadvisor_exporter metrics.
  targets    = prometheus.exporter.cadvisor.docker.targets
  forward_to = [prometheus.relabel.cadvisor_exporter.receiver]
}

prometheus.scrape "default" {
  targets = [
    {"__address__" = "demo.robustperception.io:9090", "env" = "demo"},
    {"__address__" = "prometheus.demo.do.prometheus.io", "env" = "demo"},
  ]
  job_name = "example/prometheus"
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
}

prometheus.scrape "docker_containers" {
  targets = concat(
    discovery.relabel.docker_containers.output,
  )
  forward_to = [prometheus.relabel.docker_containers.receiver]
}

prometheus.scrape "docker_local" {
  targets = [
    {"__address__" = "docker.for.mac.host.internal:9323", "env" = "host"},
    {"__address__" = "host.docker.internal:9323", "env" = "host"},
  ]
  job_name = "monitoring/docker"
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
}

prometheus.scrape "minio" {
  targets = [
    {"__address__" = "minio:9000", "app" = "minio", "container" = "minio", "namespace" = "storage", "pod" = "minio", "service" = "minio", "version" = "v1"},
  ]
  job_name = "integrations/minio"
  metrics_path = "/minio/prometheus/metrics"
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
}

prometheus.scrape "minio_v2" {
  targets = [
    {"__address__" = "minio:9000", "app" = "minio", "container" = "minio", "namespace" = "storage", "pod" = "minio", "service" = "minio", "version" = "v2"},
  ]
  job_name = "integrations/minio"
  metrics_path = "/minio/v2/metrics/cluster"
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
}

prometheus.scrape "mysql_exporter" {
  // Configure a prometheus.scrape component to collect mysql_exporter metrics.
  targets    = prometheus.exporter.mysql.grafanadb.targets
  forward_to = [prometheus.relabel.mysql_exporter.receiver]
}

prometheus.scrape "node_exporter" {
  // Configure a prometheus.scrape component to collect node_exporter metrics.
  targets    = prometheus.exporter.unix.host.targets
  forward_to = [prometheus.relabel.node_exporter.receiver]
}

prometheus.scrape "prometheus" {
  targets = [
    {"__address__" = "prometheus:9090", "app" = "prometheus", "container" = "prometheus", "pod" = "prometheus", "service" = "prometheus"},
  ]
  job_name = "prometheus"
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
}

prometheus.scrape "redis_exporter" {
  // Configure a prometheus.scrape component to collect redis_exporter metrics.
  targets    = prometheus.exporter.redis.grafana_caching.targets
  forward_to = [prometheus.relabel.redis_exporter.receiver]
}

prometheus.relabel "cadvisor_exporter" {
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
  rule {
    action        = "replace"
    replacement   = "integrations/docker"
    target_label  = "job"
  }
}

prometheus.relabel "docker_containers" {
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
  rule {
    // Drop Services that dont expose Metrics (based on App Label)
    action        = "drop"
    regex         = "grafanadb|smtp-server"
    source_labels = ["app"]
  }
  rule {
    // Drop Services that dont expose Metrics (based on Service Label)
    action        = "drop"
    regex         = "agent.*|chronograf|kapacitor|mailpit|mariadb|.*minio.*|mysql|mythical-database|.*nginx.*|postgres|provisioning-python|redis|saml-idp|smtp-server"
    source_labels = ["service"]
  }
  rule {
    // Add "group" Label to Core Infrastructure Services
    source_labels = ["container"]
    regex         = "grafana|loki|mimir|phlare|prometheus|pyroscope|tempo"
    replacement   = "infrastructure"
    target_label  = "group"
  }
  rule {
    // Add "group" Label to Mythical Beasts Services
    source_labels = ["service"]
    regex         = "mythical.*"
    replacement   = "mythical"
    target_label  = "group"
  }
}

prometheus.relabel "mysql_exporter" {
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
  rule {
    action        = "replace"
    replacement   = "grafana-mariadb-db"
    target_label  = "instance"
  }
  rule {
    action        = "replace"
    replacement   = "integrations/mysql"
    target_label  = "job"
  }
}

prometheus.relabel "node_exporter" {
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
  rule {
    action        = "replace"
    replacement   = "host"
    target_label  = "instance"
  }
  rule {
    action        = "replace"
    replacement   = "integrations/node_exporter"
    target_label  = "job"
  }
}

prometheus.relabel "redis_exporter" {
  forward_to = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
  rule {
    action        = "replace"
    replacement   = "integrations/redis_exporter"
    target_label  = "job"
  }
}

prometheus.remote_write "mimir" {
  endpoint {
    url = "http://mimir:8080/api/v1/push"
  }
  external_labels = {
    "cluster" = "docker-compose",
    "source"  = "agent-flow",
  }
}

prometheus.remote_write "prometheus" {
  endpoint {
    url = "http://prometheus:9090/api/v1/write"
  }
  external_labels = {
    "cluster" = "docker-compose",
    "source"  = "agent-flow",
  }
}

// Logs

loki.relabel "docker_containers" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/loki.relabel/
  forward_to = [loki.process.docker_containers.receiver]
  rule {
    source_labels = ["__meta_docker_container_log_stream"]
    target_label  = "stream"
  }
  rule {
    action = "labeldrop"
    regex  = "instance"
  }
}

loki.process "docker_containers" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/loki.process/
  forward_to = [loki.write.local.receiver]
  stage.match {
    selector = "{app=\"grafana\"} |= \"logger=frontend\""
    stage.logfmt {
      mapping = { "app" = "", "kind" = "" }
    }
    stage.labels {
      values = {
        app_id = "app",
        kind = "",
      }
    }
    stage.static_labels {
      values = {
        source = "faro",
      }
    }
  }
  stage.regex {
    expression = "(level=(?P<log_level>[\\s]*debug|warn|info|error|warning|trace|exception|emerg|fatal|alert|crit|critical|err|eror|information|notice|dbug))"
  }
  stage.labels {
    values = {
      level = "log_level",
    }
  }
}

loki.process "faro" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/loki.process/
  forward_to = [loki.write.local.receiver]
  stage.logfmt {
    mapping = { 
      "app_name" = "",
      "kind" = "",
    }
  }
  stage.labels {
    values = {
      app  = "app_name",
      app_id  = "app_name",
      kind = "",
    }
  }
  stage.static_labels {
    values = {
      source = "faro",
    }
  }
}

loki.source.docker "docker_containers" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/loki.source.docker/
  host          = "unix:///var/run/docker.sock"
  targets       = concat(
    discovery.relabel.docker_containers.output,
  )
  forward_to    = [loki.process.docker_containers.receiver]
  relabel_rules = loki.relabel.docker_containers.rules
}

local.file_match "nginxlogs" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/local.file_match/
  path_targets = [
    {"__path__" = "/var/log/nginx/*log", "app" = "nginx-proxy", "job" = "integrations/nginx", "namespace" = "gateway"},
  ]
}

loki.source.file "nginxlogs" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/loki.source.file/
  targets    = local.file_match.nginxlogs.targets
  forward_to = [loki.write.local.receiver]
}

local.file_match "varlogs" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/local.file_match/
  path_targets = [
    {"__path__" = "/var/log/local/{syslog,messages,*log}", "instance" = "host", "job" = "integrations/node_exporter"},
  ]
}

loki.source.file "varlogs" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/loki.source.file/
  targets    = local.file_match.varlogs.targets
  forward_to = [loki.write.local.receiver]
}

loki.write "local" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/loki.write/
  endpoint {
    url = "http://loki:3100/loki/api/v1/push"
  }
  external_labels = {
    "cluster" = "docker-compose",
    "source"  = "agent-flow",
  }
}

// Profiling
// Docs (Scrape): https://grafana.com/docs/agent/latest/flow/reference/components/pyroscope.scrape/
// Docs (Remote Write): https://grafana.com/docs/agent/latest/flow/reference/components/pyroscope.write/

pyroscope.scrape "default" {
  targets = concat(
    discovery.relabel.docker_containers.output,
  )
  forward_to      = [pyroscope.write.local.receiver]
  scrape_interval = "60s"
  scrape_timeout  = "61s" // Why does this need to be larger than the interval?
}

pyroscope.write "local" {
  endpoint {
    url = "http://pyroscope:4040"
  }
  external_labels = {
    "cluster" = "docker-compose",
    "env"     = "lab",
    "source"  = "agent-flow",
  }
}

// Traces

otelcol.receiver.jaeger "default" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.receiver.jaeger/
  protocols {
    grpc {}
    thrift_http {}
    thrift_binary {}
    thrift_compact {}
  }
  output {
    traces = [otelcol.processor.attributes.default.input]
  }
}

otelcol.receiver.otlp "default" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.receiver.otlp/
  grpc {}
  http {}
  output {
    metrics = [otelcol.processor.attributes.default.input]
    logs    = [otelcol.processor.attributes.default.input]
    traces  = [otelcol.processor.attributes.default.input]
  }
}

otelcol.processor.attributes "default" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.attributes/
  action {
    // Insert a "cluster" Attribute, only if it does not exist, and set value to "docker-compose".
    key = "cluster"
    value = "docker-compose"
    action = "insert"
  }
  action {
    // Insert a "source" Attribute, only if it does not exist, and set value to "agent-flow".
    key = "source"
    value = "agent-flow"
    action = "insert"
  }
  action {
    // Insert or Update an "env" Attribute and set value to "lab".
    key = "env"
    value = "lab"
    action = "upsert"
  }
  output {
    metrics = [otelcol.processor.batch.default.input]
    logs    = [otelcol.processor.batch.default.input]
    traces  = [otelcol.processor.batch.default.input]
  }
}

otelcol.processor.batch "default" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.processor.batch/
  output {
    metrics = [otelcol.exporter.prometheus.default.input]
    logs    = [otelcol.exporter.loki.default.input]
    traces  = [otelcol.exporter.otlp.default.input]
  }
}

otelcol.exporter.loki "default" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.exporter.loki/
  forward_to = [loki.write.local.receiver]
}

otelcol.exporter.prometheus "default" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.exporter.prometheus/
  include_scope_info   = true
  include_scope_labels = true
  include_target_info  = true
  forward_to           = [prometheus.remote_write.mimir.receiver, prometheus.remote_write.prometheus.receiver]
}

otelcol.exporter.otlp "default" {
  // https://grafana.com/docs/agent/latest/flow/reference/components/otelcol.exporter.otlp/
  client {
    endpoint = "tempo:4317"
    tls {
      insecure = true
      insecure_skip_verify = true
    } // Send traces to a locally running Tempo without TLS enabled.
  }
}
```


### Logs

```text
ob:grafana/agent log_level:debug namespace:grafana pod:agent service:agent stream:stderr]"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x523f64f]

goroutine 2953 [running]:
github.com/prometheus/node_exporter/collector.(*raplCollector).Update(0xc003033ec0, 0xc0028ab740?)
        /go/pkg/mod/github.com/grafana/node_exporter@v0.18.1-grafana-r01.0.20231004161416-702318429731/collector/rapl_linux.go:96 +0x44f
github.com/prometheus/node_exporter/collector.execute({0x75fad3d, 0x4}, {0x86481c0, 0xc003033ec0}, 0x0?, {0x863aaa0, 0xc0026eecd0})
        /go/pkg/mod/github.com/grafana/node_exporter@v0.18.1-grafana-r01.0.20231004161416-702318429731/collector/collector.go:177 +0x90
github.com/prometheus/node_exporter/collector.NodeCollector.Collect.func1({0x75fad3d?, 0x5388af?}, {0x86481c0?, 0xc003033ec0?})
        /go/pkg/mod/github.com/grafana/node_exporter@v0.18.1-grafana-r01.0.20231004161416-702318429731/collector/collector.go:168 +0x37
created by github.com/prometheus/node_exporter/collector.NodeCollector.Collect in goroutine 2948
        /go/pkg/mod/github.com/grafana/node_exporter@v0.18.1-grafana-r01.0.20231004161416-702318429731/collector/collector.go:167 +0xb4
```

--- a/pkg/integrations/node_exporter/config.go
+++ b/pkg/integrations/node_exporter/config.go
@@ -504,6 +504,10 @@ func (c *Config) mapConfigToNodeConfig() *collector.NodeCollectorConfig {
 		OldDeviceExclude: &blankString,
 	}
 
+	cfg.Rapl = collector.RaplConfig{
+		ZoneLabel: &blankBool,
+	}
+
 	cfg.Systemd = collector.SystemdConfig{
 		UnitInclude:            &c.SystemdUnitInclude,
 		UnitIncludeSet:         true,

